src/import/load-costume.js

```diff
@@ -183,9 +183,36 @@ const loadOldTextCostume = function (baseMD5ext, textMD5ext, costume, runtime) {
             ctx.drawImage(baseImageElement, 0, 0);
             ctx.drawImage(textImageElement, 0, 0);
 
-            costume.skinId = runtime.renderer.createBitmapSkin(canvas, costume.bitmapResolution, rotationCenter);
-
-            return costume;
+            return new Promise((resolve, reject) => {
+                canvas.toBlob(blob => {
+                    const reader = new FileReader();
+                    const onError = function() {
+                        // eslint-disable-next-line no-use-before-define
+                        removeEventListeners();
+                        reject();
+                    };
+                    const onLoad = function () {
+                        // eslint-disable-next-line no-use-before-define
+                        removeEventListeners();
+                        costume.assetId = runtime.storage.builtinHelper.cache(
+                            assetType,
+                            runtime.storage.DataFormat.PNG,
+                            new Buffer(reader.result)
```

Comment:
ESLint is failing because `Buffer` is not defined, but when I test this, Buffer *is* defined (the code works). Removing `Buffer` (to just pass cache `reader.result`) stops the code from working again.

Several documentation files mention taking `Buffer` objects, but (seemingly) nowhere are Buffers actually `new`ly-created. I'm not sure how I should actually construct one from my own scratch-gui code?

Reply:
Hmm... this is a bit tricky because of the differences between Node.js and the web. We've avoided the problem so far by letting `nets` and `binary-loader` create all of our buffer objects for us, but that doesn't help in your case. There's a workaround for this in `handleNewBlankCostume` but I'd rather not propagate that code elsewhere. I'm not certain it would work for this case, anyway.

The basic problem is that `Buffer` is a type in Node.js but it doesn't exist on the web. Webpack transforms `Buffer` into `Uint8Array` or similar, with a bit of wrapping code. However, `scratch-gui`'s lint settings are set to allow only browser-style code, which means there's currently no way to do this in `scratch-gui`.

I'll have to think about it a bit more, but my initial thought is to do one of these things in `scratch-storage`:
- Alter `cache` to handle more types of data, including `Blob`
- Add a way to create a new `Asset` with an empty `data` which you can then fill and provide to `cache` afterward
- Expose the `Buffer` constructor, ensuring that any code which can access `scratch-storage` can make a `Buffer` (or augmented `Uint8Array`, depending on environment and settings) of the same type that `scratch-storage` is already using.

I should be able to do one of these things next week. @rschamp (or anyone else), any thoughts on which one(s) of the above would be useful in the largest number of situations?